plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -63,6 +67,9 @@ def make_parameter(string: str) -> Any:
          - ":"-separated values for ranges of scalars.
            Eg.: "1.0:5.0"
            Note: using integers as bounds will parametrize as an integer value
+         - ":log:"-separated values for ranges of log-distributed positive scalars.
+           Eg.: "0.001:log:1.0"
+           Note: using integers as bounds will parametrize as an integer value
```

Comment:
Having the :log: there in between is a bit unusual. is this a new syntax you are inventing here or are you using it already in other places?

Reply:
new one, I feel being able to specify it is necessary, and had no idea how to do it better, so it's a start... if you have seen other options anywhere, I'm happy to change

Reply:
This is something not supported by the Ax plugin from the command line now. Logscale can only be defined in the config there.
If we want to specify types and logscale in the command line, I suggest the following syntax:

```
# Type keywords are mutually exclusive and this is enforced at runtime).
keyword: [log|float|int|str]

# 0 or more keywords, separated by :
specs: (keyword:)*
```
Does it seems good?

We can extend the Ax plugin to support it too if we have consensus on the syntax.

Reply:
As far as I am concerned, choices dont need any such spec

As mentioned in another comment, I would favor by far a float by default, and just having a `int` and a `log` spec that can be specified (and both are not exclusive, there can be log distributed integers_

Reply:
That would make this kind of combinations:
```
# bounds (min:max)
0:1           # specs are optional
int:0:1
log:0.001:1 (log 0 somehow means unbounded...)
log:int:0.001:1
int:log:0.001:1 # specs are unordered (does it matter?)

# choices
anything,whatever,something
```

Reply:
>int:log:0.001:1 # specs are unordered (does it matter?)
Yes, I don't want the command line to be sensitive to the order here.
it should just work.

no spec is going a bit wide:
maybe logscale choice does not make sense, we can issue a runtime error.

but choices might have an enforced type, especially if you default to float.

Reply:
> Yes, I don't want the command line to be sensitive to the order here.
it should just work.

That was my question, i thought you meant it was an error if it was unordered

> no spec is going a bit wide:

I don't see the problem with this, in choices we can easily infer the **value** of the choices which are provided, the problem with range is that we had to infer what was in between

Reply:
`log:int:0.001:1` the lower range is not int?

Reply:
@shagunsodhani this looks like a user error. I say throw an exception at them for this.

> I don't see the problem with this, in choices we can easily infer the value of the choices which are provided, the problem with range is that we had to infer what was in between

The problem with inferring is not just what in-between. it's also error prone.
I do agree that for choices it may not be super important to be able to enforce the type the entire set though, but it can also save some mistakes.

Reply:
> `log:int:0.001:1` the lower range is not int?

sorry, I wrote a bit too fast, would be `log:int:4:1024` for instance, and yes throwing an exception would be the way to go

> it can also save some mistakes

which ones?

Reply:
> which ones?

For example, if you have a string choice, and you have a number in the range, it won't be a string which can cause issues at runtime.
`one,1,two,2`

Reply:
This issue could be dealt with by converting to the less strict option of all, so str in this case. And if they wanted mixed types, the user can always convert back if need be.
In any case, both options could trigger exemptions in the application, which at least are expection and not silent misconfiguration. I prefer exceptions by far.

Reply:
we can delay this to when we see an actual use case.